index.js

```diff
@@ -85,49 +84,24 @@ const nonWindowsSingleCall = async (options = {}) => {
 	const lines = stdout.trim().split('\n');
 	lines.shift();
 
-	let psIndex;
-	let commPosition;
-	let argsPosition;
-
-	const processes = lines.map((line, index) => {
+	return lines.map(line => {
 		const match = psOutputRegex.exec(line);
 		if (match === null) {
 			throw new Error(ERROR_MESSAGE_PARSING_FAILED);
 		}
 
-		const {pid, ppid, uid, cpu, memory} = match.groups;
-
-		const processInfo = {
+		const {pid, ppid, uid, cpu, memory, comm, args} = match.groups;
+		const divided = args.split(' ');
+		return {
 			pid: Number.parseInt(pid, 10),
 			ppid: Number.parseInt(ppid, 10),
 			uid: Number.parseInt(uid, 10),
 			cpu: Number.parseFloat(cpu),
 			memory: Number.parseFloat(memory),
-			name: undefined,
-			cmd: undefined
+			name: comm,
+			cmd: [path.basename(divided[0]), divided.slice(1).join(' ')].join(' ')
 		};
-
-		if (processInfo.pid === psPid) {
-			psIndex = index;
-			commPosition = line.indexOf('ps', match[0].length);
-			argsPosition = line.indexOf('ps', commPosition + 2);
-		}
-
-		return processInfo;
 	});
-
-	if (psIndex === undefined || commPosition === -1 || argsPosition === -1) {
-		throw new Error(ERROR_MESSAGE_PARSING_FAILED);
-	}
-
-	const commLength = argsPosition - commPosition;
-	for (const [index, line] of lines.entries()) {
-		processes[index].name = line.slice(commPosition, commPosition + commLength).trim();
-		processes[index].cmd = line.slice(argsPosition).trim();
-	}
-
-	processes.splice(psIndex, 1);
```

Comment:
It wasn't an accident `ps` was removed from list.

Reply:
Sorry, I didn’t understand why this is. How should it work in simple words?

Reply:
Remove ps from the list.

Reply:
For now, `ps` processes removed from psList result array.

Reply:
We only need to remove the process we ran to get this list, not all `ps` processes in the system.

Reply:
Done: skip only that ps process which run for retrieving process list.